### ISSUE TITLE
printf %T should not require a parameter

### ISSUE THREAD

**Issue by dimo414** [koalaman/shellcheck#1171]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2183
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and found #775, which is closed, and #886, which doesn't appear related.

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

printf "%(%s)T"
```

#### Here's what shellcheck currently says:
```
Line 3	SC2183: This format string has 1 variables, but is passed 0 arguments.
```
#### Here's what I wanted or expected to see:

As noted in [#775](https://github.com/koalaman/shellcheck/issues/775#issuecomment-267785904) `%()T` patterns still require a parameter. At a minimum the documentation and error message should be updated to mention this requirement, though personally ShellCheck should explicitly permit the zero-parameter usage since it's documented behavior:

> If no argument is specified, conversion behaves as if -1 had been given. This is an exception to the usual printf behavior.

**Comment by koalaman** [comment]
I definitely see your point. I'm a fan of the explicit form, but for the case of a single `%T` and no other parameter, it's a bit redundant.

For now, I've just updated the docs to suggest explicitly specifying `-1`. Thanks!
